server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationClusterSettingIT.java

```diff
@@ -123,4 +133,160 @@ public void testIndexReplicationSettingOverridesDocRepClusterSetting() throws Ex
+        final int documentCount = scaledRandomIntBetween(1, 10);
+        BiConsumer<List<ReplicationType>, List<String>> consumer = (replicationList, dataNodesList) -> {
+            Settings indexSettings = Settings.builder().put(indexSettings()).put(SETTING_REPLICATION_TYPE, replicationList.get(1)).build();
+            createIndex(INDEX_NAME, indexSettings);
+        };
+        executeTest(true, consumer, INDEX_NAME, documentCount);
+    }
+
+    public void testDifferentReplicationTypes_IndexTemplates_StrictMode() throws Throwable {
+        final int documentCount = scaledRandomIntBetween(1, 10);
+
+        BiConsumer<List<ReplicationType>, List<String>> consumer = (replicationList, dataNodesList) -> {
+            client().admin()
+                .indices()
+                .preparePutTemplate("template_1")
+                .setPatterns(Collections.singletonList("test-idx*"))
+                .setSettings(Settings.builder().put(indexSettings()).put(SETTING_REPLICATION_TYPE, ReplicationType.DOCUMENT).build())
+                .setOrder(0)
+                .get();
+
+            GetIndexTemplatesResponse response = client().admin().indices().prepareGetTemplates().get();
+            assertThat(response.getIndexTemplates(), hasSize(1));
+
+            createIndex(INDEX_NAME);
+        };
+        executeTest(true, consumer, INDEX_NAME, documentCount);
+    }
+
+    public void testMismatchingReplicationType_ResizeAction_StrictMode() throws Throwable {
+        // Define resize action and target shard count.
+        List<Tuple<ResizeType, Integer>> resizeActionsList = new ArrayList<>();
+        final int initialShardCount = 2;
+        resizeActionsList.add(new Tuple<>(ResizeType.SPLIT, 2 * initialShardCount));
+        resizeActionsList.add(new Tuple<>(ResizeType.SHRINK, SHARD_COUNT));
+        resizeActionsList.add(new Tuple<>(ResizeType.CLONE, initialShardCount));
+
+        Tuple<ResizeType, Integer> resizeActionTuple = resizeActionsList.get(random().nextInt(resizeActionsList.size()));
+        final String targetIndexName = resizeActionTuple.v1().name().toLowerCase(Locale.ROOT) + "-target";
+        final int documentCount = scaledRandomIntBetween(1, 10);
+
+        logger.info("--> Performing resize action {} with shard count {}", resizeActionTuple.v1(), resizeActionTuple.v2());
+
+        // Create an index using current cluster level settings
+        BiConsumer<List<ReplicationType>, List<String>> consumer = (replicationList, dataNodesList) -> {
+            Settings indexSettings = Settings.builder()
+                .put(indexSettings())
+                .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, initialShardCount)
+                .put(SETTING_REPLICATION_TYPE, replicationList.get(0))
+                .build();
+            createIndex(INDEX_NAME, indexSettings);
+
+            for (int i = 0; i < documentCount; i++) {
+                client().prepareIndex(INDEX_NAME).setId(String.valueOf(i)).setSource("foo", "bar").get();
```

Comment:
Do you need to write any documents at all given you're just going to verify that the resize index call fails?

Reply:
Yes, indexing is not required as test expect index creation to fail. Removing this part.